src/domain/schemas/shared-schemas.js

```diff
@@ -1,9 +1,11 @@
 import joi from 'joi';
+import { slugValidationPattern } from '../validation-constants.js';
 
 export const idOrKeySchema = joi.string().alphanum().min(15).max(30);
+
 export const boolStringSchema = joi.any().valid('true', 'false', true, false);
 
-export const slugSchema = joi.string().allow('').required();
+export const slugSchema = joi.string().pattern(slugValidationPattern).allow('').required();
```

Comment:
I realize not that this might be problematic on importing old slugs 🤔

Reply:
True, I removed it for now and left a comment for re-introducing it together with https://educandu.atlassian.net/browse/EDU-308